smcontroller/smhandler.go

```diff
@@ -0,0 +1,743 @@
+			NodeID:        handler.config.NodeID,
+		}
+
+	case *pb.Alert_SystemQuotaAlert:
+		alertItem.Payload = cloudprotocol.SystemQuotaAlert{
+			Parameter: data.SystemQuotaAlert.Parameter,
+			Value:     data.SystemQuotaAlert.Value,
+			NodeID:    handler.config.NodeID,
+		}
+
+	case *pb.Alert_InstanceQuotaAlert:
+		alertItem.Payload = cloudprotocol.InstanceQuotaAlert{
+			InstanceIdent: pbconvert.NewInstanceIdentFromPB(data.InstanceQuotaAlert.Instance),
+			Parameter:     data.InstanceQuotaAlert.Parameter,
+			Value:         data.InstanceQuotaAlert.Value,
+		}
+
+	case *pb.Alert_InstanceAlert:
+		alertItem.Payload = cloudprotocol.ServiceInstanceAlert{
+			InstanceIdent: pbconvert.NewInstanceIdentFromPB(data.InstanceAlert.Instance),
+			AosVersion:    data.InstanceAlert.AosVersion,
+			Message:       data.InstanceAlert.Message,
+		}
+
+	default:
+		log.Warn("Unsupported alert alert notification")
+	}
+
+	handler.alertSender.SendAlert(alertItem)
+}
+
+func (handler *smHandler) proccLogMessage(data *pb.LogData) {
+	log.WithFields(log.Fields{
+		"id":        handler.config.NodeID,
+		"logID":     data.LogId,
+		"part":      data.Part,
+		"partCount": data.PartCount,
+	}).Debug("Receive SM push log")
+
+	if err := handler.messageSender.SendLog(cloudprotocol.PushLog{
+		LogID:     data.LogId,
+		PartCount: data.PartCount,
+		Part:      data.Part,
+		Data:      data.Data,
+		Error:     data.Error,
+	}); err != nil {
+		log.Errorf("Can't send log: %s", err)
+	}
+}
+
+func (handler *smHandler) processMonitoring(data *pb.NodeMonitoring) {
+	log.WithFields(log.Fields{"id": handler.config.NodeID}).Debug("Receive SM monitoring")
+
+	nodeMonitoring := nodeMonitoringFromPb(data)
+
+	nodeMonitoring.NodeID = handler.config.NodeID
+
+	handler.monitoringSender.SendMonitoringData(nodeMonitoring)
+}
+
+func (handler *smHandler) processOverrideEnvVarsStatus(envVarStatus *pb.OverrideEnvVarStatus) {
+	response := make([]cloudprotocol.EnvVarsInstanceStatus, len(envVarStatus.EnvVarsStatus))
+
+	for i, item := range envVarStatus.EnvVarsStatus {
+		responseItem := cloudprotocol.EnvVarsInstanceStatus{
+			InstanceFilter: cloudprotocol.NewInstanceFilter(item.Instance.ServiceId,
+				item.Instance.SubjectId, item.Instance.Instance),
+			Statuses: make([]cloudprotocol.EnvVarStatus, len(item.VarsStatus)),
+		}
+
+		for j, varStatus := range item.VarsStatus {
+			responseItem.Statuses[j] = cloudprotocol.EnvVarStatus{ID: varStatus.VarId, Error: varStatus.Error}
+		}
+
+		response[i] = responseItem
+	}
+
+	if err := handler.messageSender.SendOverrideEnvVarsStatus(
+		cloudprotocol.OverrideEnvVarsStatus{OverrideEnvVarsStatus: response}); err != nil {
+		log.Errorf("Can't sed override env ears status: %s", err.Error())
+	}
+}
+
+func (handler *smHandler) sendGetUnitConfigStatus() error {
+	if err := handler.stream.Send(
+		&pb.SMIncomingMessages{SMIncomingMessage: &pb.SMIncomingMessages_GetUnitConfigStatus{}}); err != nil {
+		return aoserrors.Wrap(err)
```

Comment:
empty line before return

Reply:
it is inside if. Not shore that it is the correct comment.